src/_P112_AS7265x.ino

```diff
@@ -0,0 +1,458 @@
+      addFormSelectorI2C(F("i2c_addr"), 1, optionValues, AS7265X_ADDR);
+      break;
+    }
+
+    case PLUGIN_SET_DEFAULTS:
+    {
+      // Set a default config here, which will be called when a plugin is assigned to a task.
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      byte   choiceMode = PCONFIG(0);
+      {
+        // sensor.setGain(AS7265X_GAIN_1X);  //Default
+        // sensor.setGain(AS7265X_GAIN_37X); //This is 3.7x
+        // sensor.setGain(AS7265X_GAIN_16X);
+        // sensor.setGain(AS7265X_GAIN_64X);
+        String optionsMode[4];
+        optionsMode[0] = F("1x");
+        optionsMode[1] = F("3.7x");
+        optionsMode[2] = F("16x");
+        optionsMode[3] = F("64x");
+        int optionValuesMode[4];
+        optionValuesMode[0] = AS7265X_GAIN_1X;
+        optionValuesMode[1] = AS7265X_GAIN_37X;
+        optionValuesMode[2] = AS7265X_GAIN_16X;
+        optionValuesMode[3] = AS7265X_GAIN_64X;
+        addFormSelector(F("Gain"), F("p112_Gain"), 4, optionsMode, optionValuesMode, choiceMode);
+      }
+      byte   choiceMode2 = PCONFIG(1);
+      {
+        // Integration cycles from 0 (2.78ms) to 255 (711ms)
+        // sensor.setIntegrationCycles(49); //Default: 50*2.8ms = 140ms per reading
+        // sensor.setIntegrationCycles(1);  //2*2.8ms = 5.6ms per reading
+        String optionsMode2[6];
+        optionsMode2[0] = F("2.8 ms");
+        optionsMode2[1] = F("28 ms");
+        optionsMode2[2] = F("56 ms");
+        optionsMode2[3] = F("140 ms");
+        optionsMode2[4] = F("280 ms");
+        optionsMode2[5] = F("711 ms");
+        int optionValuesMode2[6];
+        optionValuesMode2[0] = 0;
+        optionValuesMode2[1] = 9;
+        optionValuesMode2[2] = 19;
+        optionValuesMode2[3] = 49;
+        optionValuesMode2[4] = 99;
+        optionValuesMode2[5] = 254;
+        addFormSelector(F("Integration Time"), F("p112_IntegrationTime"), 6, optionsMode2, optionValuesMode2, choiceMode2);
+      }
+      // The status indicator (Blue LED)
+      // sensor.enableIndicator(); //Default
+      // sensor.disableIndicator();
+      addFormCheckBox(F("Blue Status LED"), F("p112_BlueStatusLED"), PCONFIG(2));
+      // sensor.takeMeasurementsWithBulb();
+      // sensor.takeMeasurements();
+      addFormCheckBox(F("Integrated UV, White, and IR LEDs"), F("p112_IntegratedLEDs"), PCONFIG(3));
+      // sensor.getCalibratedA();
+      // sensor.getA();
+      addFormCheckBox(F("Calibrated Measurements"), F("p112_CalibratedMeasurements"), PCONFIG(4));
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      PCONFIG(0) = getFormItemInt(F("p112_Gain"));
+      PCONFIG(1) = getFormItemInt(F("p112_IntegrationTime"));
+      PCONFIG(2) = isFormItemChecked(F("p112_BlueStatusLED"));
+      PCONFIG(3) = isFormItemChecked(F("p112_IntegratedLEDs"));
+      PCONFIG(4) = isFormItemChecked(F("p112_CalibratedMeasurements"));

```

Comment:
Here you use the `PCONFIG(4) to store whether or not you want to read calibrated measurements.
When you read the values, you check `PCONFIG(3)`
Maybe better to make a get/set function for it so you will not make these mistakes and you can also combine bits into a single PCONFIG entry.

For example:

```c++
bool P112_useCalibrated(struct EventStruct *event) {
  return bitRead(PCONFIG(2), 2);
}

void P112_setCalibrated(struct EventStruct *event, bool checked) {
  bitSet(PCONFIG(2), 2);
}
```

Reply:
> Here you use the `PCONFIG(4) to store whether or not you want to read calibrated measurements. When you read the values, you check `PCONFIG(3)`

You are right - Silly mistake - Already corrected